dxfeed/wrappers/subscription.py

```diff
@@ -0,0 +1,143 @@
+                                                       event_type=event_type,
+                                                       time=timestamp,
+                                                       data_len=data_len)
+
+    def __del__(self):
+        self.close_subscription()
+
+    @property
+    def event_type(self):
+        return self.__event_type
+
+    @property
+    def symbols(self):
+        return dxf_get_symbols(self.__sub)
+
+    def add_symbols(self, symbols: Union[str, Iterable[str]]):
+        """
+        Method to add symbol. Supports addition of one symbol as a string as well as list of symbols
+        Parameters
+        ----------
+        symbols: str, Iterable
+            Symbols to add. Previously added and remained symbols are ignored on C-API level
+
+        Returns
+        -------
+        self: Subscription
+        """
+        dxf_add_symbols(sc=self.__sub, symbols=cu.to_iterable_of_strings(symbols))
+        return self
+
+    def remove_symbols(self, symbols: Union[str, Iterable[str]] = None):
+        """
+        Method removes symbols from subscription. If no symbols provided removes all symbols
+
+        Parameters
+        ----------
+        symbols: str, iterable
+            One ticker or list of tickers to remove from subscription
+
+        Returns
+        -------
+        self: Subscription
+        """
+        if symbols:
+            dxf_remove_symbols(self.__sub, symbols=cu.to_iterable_of_strings(symbols))
+        else:
+            dxf_clear_symbols(self.__sub)
+        return self
+
+    def close_subscription(self):
+        """
+        Method to close subscription. All received data will remain in the object.
+        """
+        dxf_close_subscription(sc=self.__sub)
+
+    def attach_listener(self):
+        """
+        Method to attach default listener.
+
+        Returns
+        -------
+        self: Subscription
+        """
+        dxf_attach_listener(self.__sub)
+        return self
+
+    def detach_listener(self):
+        """
+        Removes listener so new events won't be received
+
+        Returns
+        -------
+        self: Subscription
+        """
+        dxf_detach_listener(self.__sub)
+        return self
+
+    def get_list(self):
+        """
+        Method to get data. When called internal field is cleared
+
+        Returns
+        -------
+        data: list
+            List of events. Each event contains list with data related to subscription type.
+        """
+        return self.__sub.get_data()
+
+    def get_dataframe(self, keep: bool = True):
+        """
+        Method converts data to the Pandas DataFrame
+
+        Parameters
+        ----------
+        keep: bool
+            When True copies data to dataframe, otherwise pops. Default True
+
+        Returns
+        -------
+        df: pandas.DataFrame
+            Dataframe each row - one event.
+        """
+        return self.__sub.to_dataframe(keep)
```

Comment:
It'll be nice to rename `to_dataframe` method to `get_dataframe` in Low-level API also.

Reply:
in 0.4.0 will be deprecated see no point in maintenance